orchestrator-bundle/orc8r-controller-operator/tests/integration/test_integration.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest  # type: ignore[import]  # noqa: F401
+
+logger = logging.getLogger(__name__)
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+
+APPLICATION_NAME = "orc8r-controller"
+CHARM_NAME = "magma-orc8r-controller-lite"
+
+
+class TestOrc8rcontroller:
+    @pytest.fixture(scope="module")
+    async def setup(self, ops_test):
+        await ops_test.model.deploy("postgresql-k8s", application_name="postgresql-k8s")
+        await ops_test.model.wait_for_idle(apps=["postgresql-k8s"], status="active", timeout=1000)
```

Comment:
As suggested by Simon in the `prometheus-edge-hub` PR. We might consider using `asyncio.gather`.
```suggestion
	await asyncio.gather(
		ops_test.model.deploy("postgresql-k8s", application_name="postgresql-k8s")
		ops_test.model.wait_for_idle(apps=["postgresql-k8s"], status="active", timeout=1000)
	)
```

Reply:
The difference was that in prometheus-edge-hub we were awaiting two independant deployments at the same time. In our case it's a bit diferent. We need to deploy before waiting for idle. There's an order to follow. The same goes for all the comments in the PR.

Reply:
Yep, my bad! Thanks for clarifying :)